src/objset.c

```diff
@@ -244,9 +244,12 @@ Int FindObjSet(Obj set, Obj obj) {
  *  Add `obj` to `set`.
  *
  *  Precondition: `set` must not already contain `obj`.
+ *
+ * inLoadWorkspace is TRUE if we are currently loading a workspace,
+ * so must not call CHANGED_BAG.
```

Comment:
How about we don't add that argument, and instead just remove the `CHANGED_BAG` calls from `AddObj*New`, and leave it to callers to invoke them afterwards? Then we need to add calls to it in `AddObjSet` and `AddObjMap`. But not in `ResizeObjSet/Map` (in fact we can remove the `CHANGED_BAG` there and replace it with a comment saying something like `SwapMasterPoint ensures both bags are marked as changed / dirty`